mlx/backend/cpu/unary.cpp

```diff
@@ -1,5 +1,8 @@
 // Copyright © 2024 Apple Inc.
 
+// Required for using M_LN2 in MSVC.
+#define _USE_MATH_DEFINES
```

Comment:
Why put it here instead of in the actual `simd/base_simd.h` file?

Reply:
`cmath` was included before `base_simd.h`, so putting it here does not work, we have to put it somewhere before any header includes `cmath`.